bin/generate-adhoc

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env ruby
```

Comment:
If possible, can we port this to Python? Not that it matters too much, but adding more languages to the code base is always a slippery slope.